apps/application-system/api/src/app/modules/application/files/file.service.spec.ts

```diff
@@ -1,122 +1,215 @@
 import { Test } from '@nestjs/testing'
 import { FileService } from './file.service'
-import { FormValue } from '@island.is/application/core'
-import {
-  ParentResidenceChange,
-  PersonResidenceChange,
-} from '@island.is/application/templates/children-residence-change'
+import { SigningService, SIGNING_OPTIONS } from '@island.is/dokobit-signing'
+import * as aws from './utils/aws'
+import * as pdf from './utils/pdf'
+import { Application } from './../application.model'
+import { ApplicationTypes, PdfTypes } from '@island.is/application/core'
+import { LoggingModule } from '@island.is/logging'
+import { NotFoundException } from '@nestjs/common'
 import {
   APPLICATION_CONFIG,
   ApplicationConfig,
 } from '../application.configuration'
 
 describe('FileService', () => {
   let service: FileService
+  let signingService: SigningService
+  const signingServiceRequestSignatureResponse = {
+    controlCode: 'code',
+    documentToken: 'token',
+  }
+
+  const bucket = 'bucket'
+
+  const applicationId = '1111-2222-3333-4444'
+
+  const parentA = {
+    nationalId: '0113215029',
+    ssn: '0113215029',
+    fullName: 'Test name',
+    phoneNumber: '111-2222',
+    email: 'email@email.is',
+  }
+
+  const parentB = {
+    id: 'id',
+    name: 'parent b',
+    ssn: '0022993322',
+    postalCode: '101',
+    address: 'Borgartún',
+    city: 'Reykjavík',
+    phoneNumber: '222-1111',
+    email: 'email2@email2.is',
+  }
+
+  const createApplication = (
+    answers: object = {
+      email: parentA.email,
+      phoneNumber: parentA.phoneNumber,
+      parentB: {
+        email: parentB.email,
+        phoneNumber: parentB.phoneNumber,
+      },
+      expiry: 'permanent',
+    },
+  ) =>
+    (({
+      id: applicationId,
+      state: 'draft',
+      applicant: parentA.ssn,
+      assignees: [],
+      typeId: ApplicationTypes.CHILDREN_RESIDENCE_CHANGE,
+      modified: new Date(),
+      created: new Date(),
+      attachments: {},
+      answers,
+      externalData: {
+        parentNationalRegistry: {
+          data: { ...parentB },
+          status: 'success',
+          date: new Date(),
+        },
+        nationalRegistry: {
+          data: { ...parentA },
+          status: 'success',
+          date: new Date(),
+        },
+      },
+    } as unknown) as Application)
 
   beforeEach(async () => {
-    const config: ApplicationConfig = {}
+    const config: ApplicationConfig = { presignBucket: bucket }
     const module = await Test.createTestingModule({
+      imports: [LoggingModule],
       providers: [
         FileService,
+        {
+          provide: SIGNING_OPTIONS,
+          useValue: {
+            url: 'Test Url',
+            accessToken: 'Test Access Token',
+          },
+        },
+        SigningService,
         { provide: APPLICATION_CONFIG, useValue: config },
       ],
     }).compile()
 
+    jest
+      .spyOn(aws, 'getFile')
+      .mockImplementation(() => Promise.resolve({ Body: 'body' }))
```

Comment:
Huh. Does this work?

Reply:
Yup do you want to go for another approach?

Reply:
Just surprised at this Jest black-magic, since I thought this module had a separate "aws" object from what is imported elsewhere.

If it works, then its fine, otherwise I would have suggested overriding it in a Nest TestingModule.